op-batcher/flags/flags.go

```diff
@@ -99,6 +99,31 @@ var (
 			return nil
 		},
 	}
+	CompressionAlgoFlag = &cli.StringFlag{
+		Name:    "compression-algo",
+		Usage:   "The compression algorithm to use. Valid options: zlib, brotli ",
+		EnvVars: prefixEnvVars("COMPRESSION_ALGO"),
+		Value:   "zlib",
+		Action: func(c *cli.Context, algo string) error {
+			if algo != "zlib" && algo != "brotli" {
+				return fmt.Errorf("unsupported compression algorithm: %s", algo)
+			}
+			return nil
+		},
+	}
+	CompressLevelFlag = &cli.IntFlag{
+		Name:    "compress-level",
+		Usage:   "The compression level to use. E.g. 1-11 for brotli, 0-9 for zlib",
+		EnvVars: prefixEnvVars("COMPRESS_LEVEL"),
+		Value:   9, // default to best zlib compression level
+		Action: func(c *cli.Context, lvl int) error {
```

Comment:
suggest dropping this action and moving this level validation code within the Check() function of optimism/op-batcher/batcher/config.go, that way you also catch config errors that don't arise from flag-based configuration.